modules/vaos/app/services/vaos/v2/eps_draft_appointment.rb

```diff
@@ -0,0 +1,461 @@
+            startBefore: Date.parse(referral.expiration_date).to_time(:utc).iso8601,
+            appointmentId: draft_appointment_id
+          }
+        )
+      end
+
+      ##
+      # Extract the first self-schedulable appointment type ID from provider
+      #
+      # Filters the provider's appointment types to find self-schedulable options
+      # and returns the ID of the first available type. Logs errors if none found.
+      #
+      # @param provider [OpenStruct] The provider containing appointment types
+      # @return [String, nil] The appointment type ID, or nil if none available
+      def get_provider_appointment_type_id(provider)
+        # Let external service BackendServiceExceptions bubble up naturally
+        if provider.appointment_types.blank?
+          Rails.logger.error("#{LOGGER_TAG}: Provider appointment types data is not available")
+          return nil
+        end
+
+        self_schedulable_types = provider.appointment_types.select { |apt| apt[:is_self_schedulable] == true }
+
+        if self_schedulable_types.blank?
+          Rails.logger.error("#{LOGGER_TAG}: No self-schedulable appointment types available for this provider")
+          return nil
+        end
+
+        self_schedulable_types.first[:id]
+      end
+
+      ##
+      # Calculate drive time from user's address to the provider
+      #
+      # Uses the user's residential address and provider's location to calculate
+      # drive time via the EPS provider service. Returns nil if user address unavailable.
+      #
+      # @param provider [OpenStruct] The provider with location coordinates
+      # @return [Hash, nil] Drive time information, or nil if user address unavailable
+      def fetch_drive_times(provider)
+        user_address = @current_user.vet360_contact_info&.residential_address
+        return nil unless user_address&.latitude && user_address.longitude
+
+        eps_provider_service.get_drive_times(
+          destinations: {
+            provider.id => {
+              latitude: provider.location[:latitude],
+              longitude: provider.location[:longitude]
+            }
+          },
+          origin: { latitude: user_address.latitude, longitude: user_address.longitude }
+        )
+      end
+
+      # =============================================================================
+      # LOGGING AND METRICS
+      # =============================================================================
+
+      ##
+      # Log metrics for referral draft creation
+      #
+      # Records StatsD metrics with sanitized referring provider and referral provider IDs
+      # for monitoring and analytics purposes.
+      #
+      # @param referral [OpenStruct, nil] The referral object containing provider information
+      # @return [void] Logs metrics to StatsD
+      def log_referral_metrics(referral)
+        return if referral.nil?
+
+        referring_provider_id = sanitize_log_value(referral.referring_facility_code)
```

Comment:
This should be the same as the other pr and be `referring_facility_code` maybe you can share log methods between the two?